File: api-dotnet/src/Application/Features/Scans/DTOs/ScanJob.cs
namespace Application.Features.Scans.DTOs;

public record ScanJob(Guid DomainId, string DomainName, Guid ScanId, string ScanType, string SurfaceType, Guid RequestedBy, DateTime EnqueuedAt)
{
    public static ScanJob Create(Guid domainId, string domainName, Guid scanId, string scanType, string surfaceType, Guid requestedBy, DateTime enqueuedAt) => new(domainId, domainName, scanId, scanType, surfaceType, requestedBy, enqueuedAt);
}

File: worker-java/src/main/java/com/vulnwatch/worker/model/ScanJob.java
package com.vulnwatch.worker.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public record ScanJob(
    @JsonProperty("DomainId")     String domainId,
    @JsonProperty("DomainName")   String domainName,
    @JsonProperty("ScanId")       String scanId,
    @JsonProperty("ScanType")     String scanType,
    @JsonProperty("SurfaceType") String surfaceType,
    @JsonProperty("RequestedBy")  String requestedBy,
    @JsonProperty("EnqueuedAt")   String enqueuedAt
) {}